Source/OxyPlot/Series/DataRange.cs

```diff
@@ -0,0 +1,111 @@
+        /// </summary>
+        /// <param name="min">lower bound.</param>
+        /// <param name="max">upper bound.</param>
+        public DataRange(double min, double max)
+        {
+            if (max < min)
+            {
+                throw new ArgumentException("max must be larger or equal min");
+            }
+
+            this.minimum = min;
+            this.maximum = max;
+        }
+
+        /// <summary>
+        /// Gets the lower bound of the data range.
+        /// </summary>
+        public double Minimum => this.minimum ?? double.NaN;
+
+        /// <summary>
+        /// Gets the upper bound of the data range.
+        /// </summary>
+        public double Maximum => this.maximum ?? double.NaN;
+
+        /// <summary>
+        /// Gets the difference between maximum and minimum.
+        /// </summary>
+        public double Range => this.Maximum - this.Minimum;
+
+        /// <summary>
+        /// Determines whether this data range is defined.
+        /// </summary>
+        /// <returns><c>false</c> if minimum or maximum is NaN, otherwise<c>true</c>.</returns>
+        public bool IsDefined()
+        {
+            return !double.IsNaN(this.Minimum) && !double.IsNaN(this.Maximum);
+        }
+
+        /// <summary>
+        /// Determines whether the specified value lies
+        /// within the closed interval of the data range.
+        /// </summary>
+        /// <param name="value">the value to be checked.</param>
+        /// <returns><c>true</c> if value in range, otherwise <c>false</c>.</returns>
+        public bool Contains(double value)
+        {
+            return value >= this.Minimum && value <= this.Maximum;
+        }
+
+        /// <summary>
+        /// Determines whether the specified data range
+        /// intersects with this instance.
+        /// </summary>
+        /// <param name="other">the other interval to be checked.</param>
+        /// <returns><c>true</c> if intersects, otherwise <c>false</c>.</returns>
+        public bool IntersectsWith(DataRange other)
+        {
+            return (this.IsDefined() && other.IsDefined()) &&
+                   (this.Contains(other.Minimum) ||
+                    this.Contains(other.Maximum) ||
+                    other.Contains(this.Minimum) ||
+                    other.Contains(this.Maximum));
+        }
+
+        /// <summary>
+        /// Returns C# code that generates this instance.
+        /// </summary>
+        /// <returns>The to code.</returns>
+        public string ToCode()
+        {
+            return CodeGenerator.FormatConstructor(this.GetType(), "{0},{1}", this.Minimum, this.Maximum);
+        }
+
+        /// <summary>
+        /// Returns a <see cref="string" /> that represents this instance.
+        /// </summary>
+        /// <returns>A <see cref="string" /> that represents this instance.</returns>
+        public override string ToString()
+        {
+            return $"{this.Minimum} {this.Maximum}";
```

Comment:
Would prefer something like$"[{min}, {max}]", so that it's clearly a closed interval (and distinguishes it from e.g. `ScreenPoint.ToString()`.

Reply:
Done.